modules/sdk-core/src/bitgo/wallet/wallets.ts

```diff
@@ -101,8 +101,21 @@ export class Wallets implements IWallets {
       throw new Error('missing required string parameter label');
     }
 
-    // no need to pass keys for (single) custodial wallets
-    if (params.type !== 'custodial') {
+    // Validate referenceWalletId parameter
+    if (params.referenceWalletId) {
+      if (!_.isString(params.referenceWalletId)) {
+        throw new Error('invalid referenceWalletId argument, expecting string');
+      }
+      if (!this.baseCoin.isEVM()) {
+        throw new Error('referenceWalletId is only supported for EVM chains');
+      }
+    }
+
+    // For wallets with referenceWalletId, skip multisig validation as configuration is inherited
+    if (params.referenceWalletId) {
+      // Skip all multisig validation - configuration will be inherited from reference wallet
+    } else if (params.type !== 'custodial') {
+      // Standard validation for non-custodial wallets without referenceWalletId
```

Comment:
We shoudn't be having a `referenceId` as a top level param that is EVM based only. Can we atleast name it `evmKeyRingReferenceWalletId`? Coin-specific logic shoudn't be leaking in generic add wallet methods, can we create a coin-specific util in eth-lib that can take care of this validation?